File: src/main/scala/ROT13.scala
class ROT13 extends IROT13 {
  override def encrypt(data: String): String = {
    //BEGIN_CHALLENGE
    val lookup = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
    val transform = "NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm"
    return data.map(x => {
      val i = transform.indexOf(x)
      if (i >= 0) lookup(i) else x
    })
    //END_CHALLENGE
    ???
  }
}

File: app/ROT13.php
<?php

function encrypt($data) {
    //BEGIN_CHALLENGE
    //return str_rot13($data); // Ohh php...
    $lookup = "\0ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
    $transform = "\0NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm";
    for($i = 0; $i < strlen($data); $i++) {
        $pos = strrpos($lookup, $data[$i]);
        if ($pos > 0)
            $data[$i] = $transform[$pos];
    }
    return $data;
    //END_CHALLENGE
}